app/main.py

```diff
@@ -1,8 +1,24 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    list_of_people = [Person(person_data["name"], person_data["age"])
+                      for person_data in people]
+
+    for person_data in people:
+        name = person_data["name"]
+        person = Person.people[name]
+        spouse = person_data.get("wife") or person_data.get("husband")
+        if spouse:
+            spouse = Person.people[spouse]
+            if "wife" in person_data:
```

Comment:
You can split the logic into 2 separate branches ( For wife and husband I'm about your `spouse`). This way your code will have less nesting and you will stop repeating yourself checking `if "wife" in person_data:`